File: src/main/java/eterea/api/rest/model/Banco.java
/**
 * Entidad de la tabla Banco
 * Revisar: Entidad sin registros, podria ser reemplazada por un enum
 */
package eterea.api.rest.model;

import lombok.*;
import org.hibernate.Hibernate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.io.Serial;
import java.io.Serializable;
import java.util.Objects;

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@Entity
@Table(name = "banco")
@AllArgsConstructor
public class Banco extends Auditable implements Serializable {

	@Serial
	private static final long serialVersionUID = -4578943377018812826L;

	@Id
	@Column(name = "ban_id")
	private Integer bancoId;
	
	@Column(name = "ban_nombre")
	private String nombre;

	@Override
	public boolean equals(Object o) {
		if (this == o) return true;
		if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
		Banco banco = (Banco) o;
		return bancoId != null && Objects.equals(bancoId, banco.bancoId);
	}

	@Override
	public int hashCode() {
		return getClass().hashCode();
	}
}

File: src/main/java/eterea/core/api/rest/kotlin/model/Banco.kt
package eterea.core.api.rest.kotlin.model

import eterea.core.api.rest.model.Auditable
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table
data class Banco(

    @Id
    @Column(name = "ban_id")
    var bancoId: Int? = null,

    @Column(name = "ban_nombre")
    var nombre: String? = null


) : Auditable()